src/Core/AdminConsole/OrganizationFeatures/OrganizationUsers/InviteUsers/InviteOrganizationUsersCommand.cs

```diff
@@ -0,0 +1,250 @@
+﻿using Bit.Core.AdminConsole.Entities;
+using Bit.Core.AdminConsole.Interfaces;
+using Bit.Core.AdminConsole.OrganizationFeatures.OrganizationUsers.InviteUsers.Models;
+using Bit.Core.AdminConsole.OrganizationFeatures.OrganizationUsers.InviteUsers.Validation;
+using Bit.Core.AdminConsole.Shared.Validation;
+using Bit.Core.Context;
+using Bit.Core.Entities;
+using Bit.Core.Enums;
+using Bit.Core.Models.Business;
+using Bit.Core.Models.Commands;
+using Bit.Core.OrganizationFeatures.OrganizationSubscriptions.Interface;
+using Bit.Core.Repositories;
+using Bit.Core.Services;
+using Bit.Core.Tools.Enums;
+using Bit.Core.Tools.Models.Business;
+using Bit.Core.Tools.Services;
+using Microsoft.Extensions.Logging;
+using static Bit.Core.AdminConsole.OrganizationFeatures.OrganizationUsers.InviteUsers.Models.CreateOrganizationUser;
+
+namespace Bit.Core.AdminConsole.OrganizationFeatures.OrganizationUsers.InviteUsers;
+
+public class InviteOrganizationUsersCommand(IEventService eventService,
+    IOrganizationUserRepository organizationUserRepository,
+    IInviteUsersValidation inviteUsersValidation,
+    IPaymentService paymentService,
+    IOrganizationRepository organizationRepository,
+    IReferenceEventService referenceEventService,
+    ICurrentContext currentContext,
+    IApplicationCacheService applicationCacheService,
+    IMailService mailService,
+    ILogger<InviteOrganizationUsersCommand> logger,
+    IUpdateSecretsManagerSubscriptionCommand updateSecretsManagerSubscriptionCommand,
+    ISendOrganizationInvitesCommand sendOrganizationInvitesCommand
+    ) : IInviteOrganizationUsersCommand
+{
+
+    public const string IssueNotifyingOwnersOfSeatLimitReached = "Error encountered notifying organization owners of seat limit reached.";
+    public const string FailedToInviteUsers = "Failed to invite user(s).";
+    public const string NoUsersToInvite = "No users to invite.";
+    public const string InvalidResultType = "Invalid result type.";
+
+    public async Task<CommandResult<ScimInviteOrganizationUsersResponse>> InviteScimOrganizationUserAsync(OrganizationUserSingleEmailInvite request)
+    {
+        var hasSecretsManager = await paymentService.HasSecretsManagerStandalone(request.InviteOrganization);
```

Comment:
On the topic of this rather expensive call. This _is_ what we discussed in our last meeting, but one of the main goals here was to optimize this, and it still bothers me that this call is so early. I've seen a pattern where an organization can't autoscale seats for another reason (e.g. it has a provider), it'll spam our SCIM endpoints with potentially hundreds of invalid requests, and we're hitting Stripe for every single one.

What if we split validation into PM validation and SM validation, and then only made this call once the PM validation passed? That is breaking the abstraction a little (we're assuming it's not required for PM validation), and we'd want to scatter some comments around, but it is a special case and would solve the problem.

Reply:
I'll have to think about this a bit. The issue is that SCIM says if you have SM you add SM to all users.

Meanwhile, you can choose whether you want to add a user to SM through the other flows. 🤷 Even if we bury it farther down, we still need to know if you have SM at the beginning to determine if the user should be added to the SM seat count.

Reply:
It would be something like:

1. check if PM subscription needs autoscaling -> validate PM subscription changes (doesn't rely on SM seat count)
2. if hasSecretsManagerStandalone -> update the request object to give everyone access to SM
3. check if SM subscription needs autoscaling -> validate SM subscription changes
4. do the invites (with the updated request object)

But this is why I say it's breaking the abstraction, you have to interpose it in the middle of the validation.

Reply:
The stripe call looks good to me, thanks for iterating on it!